File: cpp/Memory.cpp
#include <string>
#include "Product.cpp"
using namespace std;

class Memory : public Product{
    private:
        string frequency;
        int memorySize;
        bool supportCuda;

    public:
        Memory(){}

        Memory(string frequency, int memorySize, bool supportCuda){
            this->frequency = frequency;
            this->memorySize = memorySize;
            this->supportCuda = supportCuda;
        }

        void setFrequency(string frequency)
        {
            this->frequency = frequency;
        }

        string getFrequency()
        {
            return this->frequency;
        }

        void setMemorySize(int memorySize)
        {
            this->memorySize = memorySize;
        }

        int getMemorySize()
        {
            return this->memorySize;
        }

        void setSupportCuda(bool supportCuda)
        {
            this->supportCuda = supportCuda;
        }

        bool getSupportCuda()
        {
            return this->supportCuda;
        }

        ~Memory(){}
};

File: python/Memory.py
from Product import Product

class Memory(Product):
    frequency = ""
    memorySize = 0
    supportCuda = False

    def __init__(self, frequency, memorySize, supportCuda):
        self.frequency = frequency
        self.memorySize = memorySize
        self.supportCuda = supportCuda
    
    def setFrequency(self, frequency):
        self.frequency = frequency

    def getFrequency(self):
        return self.frequency

    def setMemorySize(self, memorySize):
        self.memorySize = memorySize

    def getMemorySize(self):
        return self.memorySize
    
    def setSupportCuda(self, supportCuda):
        self.supportCuda = supportCuda

    def getSupportCuda(self):
        return self.supportCuda